File: day7/puzzle1.rb
file = File.open("input")
file_data = file.read
file.close
data = file_data.split(',').map{|item| item.to_i }.sort
len = data.length
median = 0
if len.even?
    median = ((data[len / 2] + data[(len / 2) - 1]) / 2)
else
    median = data[len / 2]
end
puts "the median of all values (aka the position to go to) is:", median

fuel = 0

for i in data do
    fuel += (median - i).abs()
end
puts "total fuel cost:", fuel

File: day9/puzzle1.py
input_file = open("input", "r")
line = input_file.readline().strip()

def	get_risk_if_lowest(x, y, map):
	if y > 0 and (map[y - 1][x] <= map[y][x]):
		return 0
	if x > 0 and (map[y][x - 1] <= map[y][x]):
		return 0
	if y < (len(map) - 1) and (map[y + 1][x] <= map[y][x]):
		return 0
	if x < (len(map[y]) - 1) and (map[y][x + 1] <= map[y][x]):
		return 0
	return int(map[y][x]) + 1

map = []
while line:
	map.append(line)
	line = input_file.readline().strip()

risk = 0
y = 0
while y < len(map):
	x = 0
	while x < len(map[y]):
		risk += get_risk_if_lowest(x, y, map)
		x += 1
	y += 1
print("Risk total:", risk)
input_file.close()
